aux/inc/WireCellAux/Util.h

```diff
@@ -0,0 +1,59 @@
+/** Utilities
+ */
+
+#ifndef WIRECELLAUX_UTIL
+#define WIRECELLAUX_UTIL
+
+#include "WireCellIface/ITensorSet.h"
+#include "WireCellIface/ITensor.h"
+#include "WireCellAux/SimpleTensor.h"
+#include "WireCellUtil/Exceptions.h"
+
+#include <Eigen/Core>
+
+#include <iostream>
+
+namespace WireCell
+{
+namespace Aux
+{
+
+std::string dump(const ITensor::pointer iten);
+
+template<typename ElementType>
+inline Eigen::Array<ElementType, Eigen::Dynamic, Eigen::Dynamic> itensor_to_eigen_array(const ITensor::pointer iten)
+{
+    if (!iten)
+    {
+        THROW(ValueError() << errmsg{"(iten == null"});
+    }
+
+    if (iten->shape().size() != 2)
```

Comment:
ITensors can be arbitrary shape and Eigen matrices and arrays can be 1D or 2D and Eigen tensors can be ND (these are in Eigen-unsupported).

I wonder if this converter can be made to handle at least 1D or 2D

If the exact signature can't handle this diversity we can make different template specializations.

This comment also applies to the inverse function below.

Reply:
Will try to improve this in future versions.